src/Input/Input.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Minicli\Framework\Input;
+
+final class Input
+{
+    private const DEFAULT_COMMAND = 'default';
+
+    /**
+     * The command to be called.
+     */
+    private string $command = '';
+
+    /**
+     * The raw arguments passed to the command.
+     *
+     * @var array<int,string> $rawArgs
+     */
+    private array $rawArgs = [];
+
+    /**
+     * The raw arguments passed to the command.
+     *
+     * @var array<int,string> $args
+     */
+    private array $args = [];
+
+    /**
+     * The parsed parameters passed to the command.
+     *
+     * @var array<string,string> $params
+     */
+    private array $params = [];
+
+    /**
+     * The parsed flags passed to the command.
+     *
+     * @var array<int, string> $flags
+     */
+    private array $flags = [];
+
+    public function __construct(array $argv)
+    {
+        $this->rawArgs = $argv;
+
+        $this->parseInput();
+
+        if ($this->args[1]) {
+            $this->command = $this->args[1];
```

Comment:
Could we not simplify this?

```php
$this->command = $this->args[1] ?? [];
```

Reply:
Sure
As soon as possible I’ll update it